compiler/packages/babel-plugin-react-compiler/src/Optimization/ConstantPropagation.ts

```diff
@@ -154,6 +156,46 @@ function applyConstantPropagation(
         }
         break;
       }
+      case 'ternary': {
+        if (!fn.env.config.enableTernaryConstantPropagation) {
+          break;
+        }
+        const branchBlock = fn.body.blocks.get(terminal.test);
+        if (branchBlock === undefined) {
+          break;
+        }
+
+        if (branchBlock.terminal.kind !== 'branch') {
+          // TODO: could be other kinds like logical
+          break;
+        }
+
+        const testValue = read(constants, branchBlock.terminal.test);
+
+        if (testValue !== null && testValue.kind === 'Primitive') {
+          hasChanges = true;
+          const targetBlockId = testValue.value
+            ? /*
+               * Do I need to change these from being value blocks?
+               * Currently getting
+               * Invariant: Expected a fallthrough for value block (6:6)
+               */
+              branchBlock.terminal.consequent
+            : branchBlock.terminal.alternate;
+          block.terminal = {
+            kind: 'goto',
+            variant: GotoVariant.Break,
+            block: targetBlockId,
+            id: terminal.id,
+            loc: terminal.loc,
+          };
+
+          fn.body.blocks.get(branchBlock.terminal.consequent)!.kind = 'block';
+          fn.body.blocks.get(branchBlock.terminal.alternate)!.kind = 'block';
```

Comment:
This assumes that the consequent block doesn't have further nested value blocks "within" it, which would also need to be converted to value blocks. Simplest thing to do is to add a check that the target block ends with a goto to the fallthrough of the outer ternary